src/components/Auth.jsx

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect, useState } from "react";
+import "../styles/login.css";
+import "../styles/signup.css";
+import btnOffImg from "../assets/btn_visibility_off_24px.png";
+import btnOnImg from "../assets/btn_visibility_on_24px.png";
+import googleLogoImg from "../assets/google_logo.png";
+import kakaoLogoImg from "../assets/kakao_logo.png";
+import Logo from "./Logo";
+import { validEmail, validNickname, validPassword } from "../utils/valid";
+import { useNavigate } from "react-router-dom";
+
+const Email = ({ value, setValue, isValid, setIsValid }) => {
+  const [message, setMessage] = useState("");
+
+  const focusoutAuthInput = ({ target }) => {
+    setMessage(validEmail(target));
+    setIsValid(validEmail(target) === "" && value !== "");
+  };
+
+  return (
+    <div className="auth__el">
+      <label className="auth__label" htmlFor="email">
+        이메일
+      </label>
+      <input
+        className={`auth__input ${
+          !isValid && message !== "" ? "auth__invalid" : ""
+        }`}
+        type="email"
+        id="email"
+        placeholder="이메일을 입력해주세요"
+        required
+        value={value}
+        onBlur={focusoutAuthInput}
+        onChange={({ target }) => setValue(target.value)}
+      />
+      <span
+        className={isValid ? "hidden" : ""}
+        style={{
+          padding: "8px 16px 0px",
+          color: "#f74747",
+        }}
+      >
+        {message}
+      </span>
+    </div>
+  );
+};
+
+const NickName = ({ isValid, setIsValid }) => {
+  const [value, setValue] = useState("");
+  const [message, setMessage] = useState("");
+
+  const focusoutAuthInput = ({ target }) => {
+    setMessage(validNickname(target));
+    setIsValid(validNickname(target) === "" && value !== "");
+  };
+
+  return (
+    <div className="auth__el">
+      <label className="auth__label" htmlFor="nickname">
+        닉네임
+      </label>
+      <input
+        className={`auth__input ${
+          !isValid && message !== "" ? "auth__invalid" : ""
+        }`}
+        type="text"
+        id="nickname"
+        placeholder="닉네임을 입력해주세요"
+        required
+        value={value}
+        onBlur={focusoutAuthInput}
+        onChange={({ target }) => setValue(target.value)}
+      />
+      <span
+        className={isValid ? "hidden" : ""}
+        style={{
+          padding: "8px 16px 0px",
+          color: "#f74747",
+        }}
+      >
+        {message}
+      </span>
+    </div>
+  );
+};
+
+const Password = ({ isValid, setIsValid }) => {
+  const [value, setValue] = useState("");
+  const [message, setMessage] = useState("");
+
+  const focusoutAuthInput = ({ target }) => {
+    setMessage(validPassword(target));
+    setIsValid(validPassword(target) === "" && value !== "");
+  };
+
+  const keyupCheckPasswordSame = ({ target }) => {};
```

Comment:
```suggestion
  const keyupCheckPasswordSame = ({ target }) => {};
```
함수 본문에 액션이 없는데 혹시 만들다가 살짝 스킾 된 함수인가요??

Reply:
네 password 확인 과정을 부모 컴포넌트로 올렸습니다